chapter.md

```diff
@@ -0,0 +1,735 @@
+and bind its inputs to objects retrieved from the Redux state.
+The first part of that form looks like this:
+
+```html
+<form #form="ngForm">
+  <label for="name">Character Name:</label>
+  <input
+    type="text"
+    name="name"
+    [(ngModel)]="characterForm.name">
+
+  <label for="age">Age:</label>
+  <input
+    type="number"
+    name="age"
+    [(ngModel)]="characterForm.bioSummary.age">
+</form>
+```
+
+`ngModel` is a magic word meaning "the value of this field in the form".
+The expression `[(ngModel)]` therefore makes a form control
+and binds it to the public data member of our form object.
+The final step in wiring all of this together is therefore to subscribe to the Redux state
+so that whenever it changes,
+the changes are automatically pushed into the form
+(which triggers a DOM update and a re-rendering of the page).
+Equally,
+since we're listening to the form for changes and writing those to our state
+by creating and dispatching actions,
+anything the user does will be reflected in the state.
+The beauty of this is that if anyone else does an update anywhere,
+everything will keep itself in sync.
+
+It's tempting to put this final piece of wiring in the constructor of our component,
+but that doesn't work
+because Angular has to construct several objects
+before any of them can be wired together.
+This is a key feature of Angular's architecture
+(and of the architectures of many other systems):
+object *construction* and object *initialization* are handled separately
+to free us from headaches related to cyclic references.
+
+The right place to connect everything is `ngOnInit`,
+which is called after all the objects in the system have been created
+but before any of them are used.
+The `ngOnInit` for `CharacterForm` does the two pieces of wiring described above
+in an arbitrary order:
+
+```ts
+ngOnInit() {
+
+  // Subscribe to the form in state.
+  this.formSubs = this.ngRedux.select(state => state.form.character)
+    .subscribe(characterFormState => {
+      this.characterForm = characterFormState;
+    });
+
+  // Dispatch an action when the form is changed.
+  this.ngForm.valueChanges.debounceTime(0)
+    .subscribe(change =>
+      this.ngRedux.dispatch(
+        saveForm(
+          change,
+          ['character']
+        )
+      )
+    );
+}
+```
+
+The first half of this method says that when the state changes,
+we want to update the form.
+We use `ngRedux.select` with a fat arrow function as callback
+to pick out the character portion of the state,
+then subscribe to that with a another fat arrow callback
+that updates the form data
+(which as explained above triggers redisplay).
+
+The second half of `ngOnInit` does the binding in the opposite direction:
+whenever the form changes,
+we dispatch an action created by `saveForm`
+that has `SAVE_FORM` as the change
+and `['character']` as the path to the part of the state we want to modify.
+FIXME: how does `SAVE_FORM` get into this as `change`?
```

Comment:
when you subscribe to `ngForm.valueChanges` the callback function you give to it will be called with an object that represents the form
this object due the names we give in the template matches exactly the form structure we use on the state